compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -326,6 +327,23 @@ pub unsafe fn create_module<'ll>(
         llvm::LLVMMDNodeInContext(llcx, &name_metadata, 1),
     );
 
+    // Emit RISC-V specific target-abi metadata
+    // to workaround lld as the LTO plugin not
+    // correctly setting target-abi for the LTO object
+    // FIXME: https://github.com/llvm/llvm-project/issues/50591
+    // If llvm_abiname is empty, emit nothing.
+    if matches!(sess.target.arch.as_ref(), "riscv32" | "riscv64")
+        && !sess.target.options.llvm_abiname.is_empty()
+    {
+        let llvm_abiname = CString::new(sess.target.options.llvm_abiname.as_ref()).unwrap();
```

Comment:
instead of creating a new `cstr`, I think we can follow other parts to pass `value` and `valueLen` to create a StringRef

Reply:
Thanks for your review! This is clearly a better way.